customer-datastore-inmemory/src/main/java/com/telecominfraproject/wlan/customer/datastore/inmemory/CustomerDatastoreInMemory.java

```diff
@@ -175,6 +180,114 @@ public Customer delete(int id) {
 
         return result;
     }
+    
+    @Override
+    public PaginationResponse<Customer> searchAll(String criteria, String username,
+    		final List<ColumnAndSort> sortBy, PaginationContext<Customer> context) {
+    	
+    	if (context == null) {
+    		context = new PaginationContext<>();
+    	}
+    	
+    	PaginationResponse<Customer> ret = new PaginationResponse<>();
+    	ret.setContext(context.clone());
+    	
+    	if (ret.getContext().isLastPage()) {
+    		// no more pages available according to the context
+    		return ret;
+    	}
+    	
+        List<Customer> items = new LinkedList<>();
+        
+        if (username == null) {
+	        for (Customer mdl : idToCustomerMap.values()) {
+	        	if (mdl.getName() != null && 
+	        			mdl.getName().toLowerCase().contains(criteria.toLowerCase()) || 
+	        			mdl.getEmail().toLowerCase().contains(criteria.toLowerCase())) {
+	        		items.add(mdl);
+	        	}
+	        }
+        } else {
+        	for (Customer mdl : idToCustomerMap.values()) {
+	        	if (mdl.getName() != null && 
+	        			mdl.getName().toLowerCase().contains(criteria.toLowerCase()) && 
+	        			mdl.getEmail().toLowerCase().equals(username)) {
```

Comment:
username should be matched to the usernames of the portal users that are related to the customer. 
which kind of forces the customer datastore and portal user datastores to be co-located in the same DB.